http-streaming/README.md

```diff
@@ -0,0 +1,297 @@
+
+* JSON-based REST API (there are also Python and Java clients exposing the REST API)
+* The data returned is state rather than discrete events. For example, the [State Vectors](https://openskynetwork.github.io/opensky-api/rest.html#all-state-vectors) method that this demo builds on returns aircraft state (e.g., location, altitude, and speed) as of the time of the request.
+* The data needs a bit of massaging and cleaning. In the case of this demo, adding schema and cleaning string fields are a couple of the ways that the data is shaped and cleaned on its way to becoming a Kafka-based stream.
+
+## Prerequisites
+
+The following tools are required to run this demo:
+
+* A Confluent Cloud account. [Sign up](https://www.confluent.io/confluent-cloud/tryfree) for a free trial if you don't already have one.
+* The Confluent CLI. Refer to the installation instructions [here](https://docs.confluent.io/confluent-cli/current/install.html).
+* The `confluent-flink-quickstart` CLI plugin. This plugin spins up the Kafka and Flink resources in Confluent Cloud that are needed for this demo. Install the plugin by running:
+  ```bash
+  confluent plugin install confluent-flink-quickstart
+  ```
+* Review the OpenSky Network's [terms of use](https://opensky-network.org/about/terms-of-use).
+
+## Provision Confluent Cloud infrastructure
+
+We will use the Confluent CLI to provision the Confluent Cloud resources used in this demo.
+
+First, login via the CLI
+
+```bash
+confluent login --prompt --save
+```
+
+Next, run the following command to spin up a Kafka cluster and Flink compute pool in Confluent Cloud:
+
+```bash
+confluent flink quickstart \
+    --name http-streaming \
+    --max-cfu 10 \
+    --region us-east-1 \
+    --cloud aws
+```
+
+The command will run for about a minute and drop you into an interactive Flink shell.
+
+Once you're in the Flink shell, run your first Flink SQL statement to see what Flink SQL tables exist. In the Flink shell:
+
+```sql
+show tables;
+```
+
+Since we don't have any data in the system yet, you will see the message:
+
+```noformat
+The server returned empty rows for this statement.
+```
+
+Don't worry! This is expected since we haven't produced any data into the system yet.
+
+## Launch HTTP Source Connector
+
+Run the `scripts/create-connector.sh` script included in this repository to provision an [HTTP Source Connector](https://docs.confluent.io/cloud/current/connectors/cc-http-source.html) in Confluent Cloud. The script will complete in a few seconds, and then the resulting connector will need a minute or so to be provisioned.
+
+Note the following connector configurations in the script:
+
+We specify that we are consuming the OpenSky Network's [All State Vectors](https://openskynetwork.github.io/opensky-api/rest.html#all-state-vectors) method in the `url` property:
+```noformat
+"url": "https://opensky-network.org/api/states/all"
+```
+
+The `request.interval.ms` property specifies how often we should poll the API:
```

Comment:
this would be something I'd be interested in -- what are best practices for polling REST APIs? Just scraping under the rate limit or is there something else to consider?

Reply:
added a tangent on this